src/sync/generic_network_transport.hpp

```diff
@@ -0,0 +1,335 @@
+        else if (code == "ServiceAlreadyExists")
+            return ServiceErrorCode::service_already_exists;
+        else if (code == "ServiceCommandNotFound")
+            return ServiceErrorCode::service_command_not_found;
+        else if (code == "ValueNotFound")
+            return ServiceErrorCode::value_not_found;
+        else if (code == "ValueAlreadyExists")
+            return ServiceErrorCode::value_already_exists;
+        else if (code == "ValueDuplicateName")
+            return ServiceErrorCode::value_duplicate_name;
+        else if (code == "FunctionNotFound")
+            return ServiceErrorCode::function_not_found;
+        else if (code == "FunctionAlreadyExists")
+            return ServiceErrorCode::function_already_exists;
+        else if (code == "FunctionDuplicateName")
+            return ServiceErrorCode::function_duplicate_name;
+        else if (code == "FunctionSyntaxError")
+            return ServiceErrorCode::function_syntax_error;
+        else if (code == "FunctionInvalid")
+            return ServiceErrorCode::function_invalid;
+        else if (code == "IncomingWebhookNotFound")
+            return ServiceErrorCode::incoming_webhook_not_found;
+        else if (code == "IncomingWebhookAlreadyExists")
+            return ServiceErrorCode::incoming_webhook_already_exists;
+        else if (code == "IncomingWebhookDuplicateName")
+            return ServiceErrorCode::incoming_webhook_duplicate_name;
+        else if (code == "RuleNotFound")
+            return ServiceErrorCode::rule_not_found;
+        else if (code == "APIKeyNotFound")
+            return ServiceErrorCode::api_key_not_found;
+        else if (code == "RuleAlreadyExists")
+            return ServiceErrorCode::rule_already_exists;
+        else if (code == "AuthProviderDuplicateName")
+            return ServiceErrorCode::auth_provider_duplicate_name;
+        else if (code == "RestrictedHost")
+            return ServiceErrorCode::restricted_host;
+        else if (code == "APIKeyAlreadyExists")
+            return ServiceErrorCode::api_key_already_exists;
+        else if (code == "IncomingWebhookAuthFailed")
+            return ServiceErrorCode::incoming_webhook_auth_failed;
+        else if (code == "ExecutionTimeLimitExceeded")
+            return ServiceErrorCode::execution_time_limit_exceeded;
+        else if (code == "NotCallable")
+            return ServiceErrorCode::not_callable;
+        else if (code == "UserAlreadyConfirmed")
+            return ServiceErrorCode::user_already_confirmed;
+        else if (code == "UserNotFound")
+            return ServiceErrorCode::user_not_found;
+        else if (code == "UserDisabled")
+            return ServiceErrorCode::user_disabled;
+        else
+            return ServiceErrorCode::unknown;
+    }
+
+    const ServiceErrorCode m_code;
+    const std::string m_raw_code;
+};
+
+} // namespace error
+
+
+enum class Method {
+    get, post, patch
```

Comment:
Any reason `delete` and `put` is not here?
Also, please add documentation to all public structures.

Reply:
Also, I would consider renaming this to `HttpMethod` for clarity

Reply:
Seemed pertinent to only add methods as needed.

Reply:
I just assumed that the first one attempting to add a PUT or DELETE would get rather annoyed if it is missing, but it isn't something I feel strongly about.